sdk/containerregistry/azure-containerregistry/tests/test_container_repository_client_async.py

```diff
@@ -168,3 +163,76 @@ async def test_set_manifest_properties_does_not_exist(self, containerregistry_ba
 
         with pytest.raises(ResourceNotFoundError):
             await client.set_manifest_properties("sha256:abcdef", ContentPermissions(can_delete=False))
+
+    @acr_preparer()
+    async def test_list_tags_descending(self, containerregistry_baseurl):
+        client = self.create_repository_client(containerregistry_baseurl, self.repository)
+
+        prev_last_updated_on = None
+        count = 0
+        async for tag in client.list_tags(order_by=TagOrderBy.LAST_UPDATE_TIME_DESCENDING):
+            if prev_last_updated_on:
+                assert tag.last_updated_on < prev_last_updated_on
+            prev_last_updated_on = tag.last_updated_on
+            count += 1
+
+        assert count > 0
+
+    @acr_preparer()
+    async def test_list_tags_ascending(self, containerregistry_baseurl):
+        client = self.create_repository_client(containerregistry_baseurl, self.repository)
+
+        prev_last_updated_on = None
+        count = 0
+        async for tag in client.list_tags(order_by=TagOrderBy.LAST_UPDATE_TIME_ASCENDING):
+            if prev_last_updated_on:
+                assert tag.last_updated_on > prev_last_updated_on
+            prev_last_updated_on = tag.last_updated_on
+            count += 1
+
+        assert count > 0
+
+
+    @acr_preparer()
+    async def test_list_registry_artifacts(self, containerregistry_baseurl):
+        client = self.create_repository_client(containerregistry_baseurl, self.repository)
+
+        count = 0
+        async for artifact in client.list_registry_artifacts():
+            assert artifact is not None
+            assert isinstance(artifact, RegistryArtifactProperties)
```

Comment:
is there a reason for checking `not None` before checking `isinstance`? I would think `isinstance` assertion would encompass `not None`?

Reply:
You could absolutely do this. Mostly a matter of habit.